---
uint8 code
string message
